src/builder/List.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.23;
+
+import {IQuarkWallet} from "quark-core/src/interfaces/IQuarkWallet.sol";
+import {Actions} from "./Actions.sol";
+
+library List {
+    error IndexOutOfBound();
+
+    struct ListStuct {
```

Comment:
Typo in `Struct`. But also, consider renaming this as `DynamicArray` since `List` is really generic 

```
    struct DynamicArray {
        bytes[] items;
        uint256 length;
    }
```

Reply:
Note: `size` is a bit confusing because this is really the length of the array. The size is determined by `items.length`.

Reply:
Sure no problem 👍 
Oh yeah I think the reason I chose `size` initially was borrowing it from Java's Arraylist convention on size() haha. 
We can use `length` 😄